x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizerTests.java

```diff
@@ -925,21 +925,12 @@ public void testMissingFieldsDoNotGetExtracted() {
                 "salary"
             )
         );
-        // emp_no
-        assertThat(projections.get(1), instanceOf(ReferenceAttribute.class));
-        // first_name
-        assertThat(projections.get(2), instanceOf(ReferenceAttribute.class));
-
-        // last_name --> first_name
-        var nullAlias = Alias.unwrap(projections.get(8));
-        assertThat(Expressions.name(nullAlias), is("first_name"));
-        // salary --> emp_no
-        nullAlias = Alias.unwrap(projections.get(10));
-        assertThat(Expressions.name(nullAlias), is("emp_no"));
-        // check field extraction is skipped and that evaled fields are not extracted anymore
         var field = as(project.child(), FieldExtractExec.class);
         var fields = field.attributesToExtract();
         assertThat(Expressions.names(fields), contains("_meta_field", "gender", "hire_date", "job", "job.raw", "languages", "long_noidx"));
+        var eval = as(field.child(), EvalExec.class);
```

Comment:
Could you also, please, change the "Expects" part of this method at the top? It's a thing with the optimizer tests that, at least for me, is very useful when looking at a test.

Reply:
I've taken a different approach, so we no longer need to adjust this test.